src/main/java/org/breedinginsight/brapps/importer/model/imports/germplasm/GermplasmImportService.java

```diff
@@ -227,15 +243,34 @@ public ImportPreviewResponse process(List<BrAPIImport> brAPIImports, Table data,
         ));
         response.setRows((List<PendingImport>)(List<?>) mappedBrAPIImport);
 
+        // Save our results to the db
+        String json = null;
+        try {
+            json = objMapper.writeValueAsString(response);
+        } catch(JsonProcessingException e) {
+            log.error("Problem converting mapping to json", e);
+            // If we didn't catch this error in the validator, this is an unexpected server error.
+            throw new InternalServerException("Problem converting mapping to json", e);
```

Comment:
Should a status be written to the database when this happens?

Reply:
Exceptions are caught in the `FileImportService` to be written to the db:

```
CompletableFuture.supplyAsync(() -> {
            try {
                importService.process(brAPIImportList, data, program, upload, commit);
            } catch (UnprocessableEntityException e){
                ImportProgress progress = upload.getProgress();
                progress.setStatuscode((short) HttpStatus.UNPROCESSABLE_ENTITY.getCode());
                progress.setMessage(e.getMessage());
                progress.setUpdatedBy(actingUser.getId());
                importDAO.update(upload);
            } catch (DoesNotExistException e)  {
                ImportProgress progress = upload.getProgress();
                progress.setStatuscode((short) HttpStatus.NOT_FOUND.getCode());
                progress.setMessage(e.getMessage());
                progress.setUpdatedBy(actingUser.getId());
                importDAO.update(upload);
            } catch (Exception e) {
                ImportProgress progress = upload.getProgress();
                progress.setStatuscode((short) HttpStatus.INTERNAL_SERVER_ERROR.getCode());
                // TODO: Probably don't want to return this message. But do it for now
                progress.setMessage(e.getMessage());
                progress.setUpdatedBy(actingUser.getId());
                importDAO.update(upload);
            }
            return null;
        });
```

Reply:
Ah, thanks for pointing that out!